src/backend/src/services/slack.services.ts

```diff
@@ -0,0 +1,289 @@
+    }
+
+    //loop through the blocks of the meta data while accumulating the
+    //text and users notified
+    let messageText = '';
+    let userIdsToNotify: string[] = [];
+
+    //Get the settings of all users in this organization to compare slack ids
+    const users = await UsersService.getAllUsers();
+    const userSettings = await Promise.all(
+      users.map((user) => {
+        return UsersService.getUserSettings(user.userId);
+      })
+    );
+
+    //get the name of the user that sent the message from slack
+    let userName: string = '';
+    try {
+      userName = (await getUserName(eventMessage.user)) ?? '';
+    } catch (ignored) {}
+
+    //if slack could not produce the name of the user, look for their name in prisma
+    if (!userName) {
+      const userIdList = userSettings
+        .filter((userSetting) => userSetting.slackId === eventMessage.user)
+        .map((userSettings) => userSettings.userId);
+      if (userIdList.length !== 0) {
+        const prismaUserName = users.find((user) => user.userId === userIdList[0]);
+        userName = prismaUserName
+          ? `${prismaUserName?.firstName} ${prismaUserName?.lastName}`
+          : 'Unknown User:' + eventMessage.user;
+      } else {
+        userName = 'Unknown_User:' + eventMessage.user;
+      }
+    }
+
+    //pull out the blocks of data from the metadata within the message event
+    const richTextBlocks = eventMessage.blocks?.filter((eventBlock: any) => eventBlock.type === 'rich_text');
+
+    if (richTextBlocks && richTextBlocks.length === 1) {
+      for (const element of richTextBlocks[0].elements[0].elements) {
+        messageText += await slackServices.blockToString(element);
+        userIdsToNotify = userIdsToNotify.concat(
+          await slackServices.blockToMentionedUsers(element, userSettings, event.channel)
+        );
+      }
+    } else {
+      return;
+    }
+
+    //get rid of duplicates within the users to notify
+    userIdsToNotify = [...new Set(userIdsToNotify)];
+
+    //if no users are notified, disregard the message
+    if (userIdsToNotify.length === 0) {
+      return;
+    }
+
+    console.log('processed event');
+
+    if (event.subtype === 'message_changed') {
+      //try to edit the announcement, if no announcement with that id exists create a new announcement
+      try {
+        return await AnnouncementService.updateAnnouncement(
+          messageText,
+          userIdsToNotify,
+          dateCreated,
+          userName,
+          eventMessage.client_msg_id,
+          slackChannelName,
+          organizationId
+        );
+      } catch (ignored) {}
+    }
+    try {
+      return await AnnouncementService.createAnnouncement(
+        messageText,
+        userIdsToNotify,
+        dateCreated,
+        userName,
+        eventMessage.client_msg_id,
+        slackChannelName,
+        organizationId
+      );
+    } catch (error) {
+      //if announcement does not have unique cient_msg_id disregard it
```

Comment:
this doesnt have to be the case why this statement fails, but we should get rid of this try catch anyways